### DISCUSSION TITLE
Reaction works even without making [Atom]

### DISCUSSION THREAD

**Discussion by slimshader** [codewriter-packages/UniMob#32] [General]
Hi,

is this a desired behaviour:

```
    [Atom]
    public string Name { get; set; }

    public bool IsNameValid => !string.IsNullOrEmpty(Name) && Name.Length > 3;
```

later:
```
        Atom.Reaction(Lifetime, () => _user.IsNameValid, x =>
        {
            Debug.Log($"IsNameValid: {x}");
            _nameLabel.style.color = x ? Color.green : Color.red;
        });
```

reaction is properly reacting even tho IsNameValid is not marked as [Atom].

If it is desired, then do I understand correcly that:

1. Reaction works because it registeres to underlying Name atom
2. because IsNameValid is not marked as [Atom] the result is not cashed for that property

**Comment by vanifatovvlad** [comment]
Yes, this is intended behavior

- Right. Reactions are subscribed to every atom that have been read. In this case it is the `Name`. It does not matter how the atoms were read, directly or through multiple nested method calls
- Right. This is a regilar C# property and, accordingly, without a cache

For example, here if you do not specify the `Atom` attribute, the error log will be printed every second. With the attribute it will be printed only once since the result will be cached

```csharp
using UniMob;
using UnityEngine;

public class AtomCaching : LifetimeMonoBehaviour
{
    [Atom] public int A { get; set; }
    [Atom] public int B { get; set; }

    /*[Atom]*/ public int Computed
    {
        get
        {
            Debug.LogError("Compute");
            return A;
        }
    }

    private void Awake()
    {
        Atom.Reaction(Lifetime, () => Debug.Log(Computed + B));

        InvokeRepeating(nameof(IncrementB), 1f, 1f);
    }

    private void IncrementB()
    {
        B++;
    }
}
```

**Reply by slimshader** [reply]
I see, thanks againt for great explanation.

btw, UniMob starts to become a crucial library for me along UniTask and vContainer, I see a long future ahead ;)
